docs/components/content/toc-context.tsx

```diff
@@ -0,0 +1,66 @@
+import type { HeadingProps } from '@spark-web/heading';
+import { Heading as SparkHeading } from '@spark-web/heading';
+import { HEADER_HEIGHT } from 'components/constants';
+import type { ProviderProps } from 'react';
+import { createContext, useContext, useMemo } from 'react';
+
+import type { HeadingData } from '../../utils/generate-toc';
+import {
+  IntersectionObserverProvider,
+  useEntries,
+  useObserve,
+} from './intersection-observer';
+
+const TocContext = createContext<HeadingData[] | undefined>(undefined);
+
+export const TocContextProvider = (props: ProviderProps<HeadingData[]>) => (
+  <IntersectionObserverProvider
+    options={{ rootMargin: `-${HEADER_HEIGHT}px 0px 0px 0px` }}
+  >
+    <TocContext.Provider {...props} />
+  </IntersectionObserverProvider>
+);
+
+export const useTocContext = (): {
+  headings: HeadingData[];
+} => {
+  const context = useContext(TocContext);
+  if (typeof context === 'undefined') {
+    throw new Error('useTocContext must be used within a TocContextProvider');
+  }
+  return { headings: context };
+};
```

Comment:
Suggestion, rename `useTocContext` to `useHeadings` and create a generic hook that throws for us if context is undefined to save duplicate code:
```suggestion
export const useTocContext = () => {
  const context = useContext(TocContext);
  if (typeof context === 'undefined') {
    throw new Error('useTocContext must be used within a TocContextProvider');
  }
  return context;
};

export const useHeadings = () => {
  const headings = useTocContext();
  return { headings };
};
```

Reply:
but then the next function will become:
```
export const Heading = (props: HeadingProps) => {
  useTocContext();
  ...
};
```
which doesn't actually use the context, and only calls the hook to throw the error, but someone else might think that it's not being used and delete it.
Also, the error message will say `useTocContext must be used within a TocContextProvider` as opposed to `Heading must be used within a TocContextProvider`

Reply:
Couldn't we rely on `useObserve` throwing the error?
That way we'd only need:
```tsx
export const Heading = (props: HeadingProps) => {
  const { ref } = useObserve<HTMLHeadingElement>();
  return <SparkHeading ref={ref} {...props} />;
};
```

Reply:
it does throw an error, but the error message will be `useObserve must be used within a IntersectionObserverProvider` as opposed to `Heading must be used within a TocContextProvider`

Reply:
Maybe the clearer error messages are worth the extra lines of code, IDK ¯\_(ツ)_/¯ 
PR looks good, I've approved so I'll let you decide 😊

Reply:
> it does throw an error, but the error message will be useObserve must be used within a IntersectionObserverProvider as opposed to Heading must be used within a TocContextProvider
🤦‍♂️ very good, carry on, nothing to see here.

Reply:
🙈